### PR TITLE
issue: 846410 Add events support for vma_poll()

### PR DESCRIPTION
**Pull request by igor-ivanov**
These changes should extend vma_poll() capabilities to return events in vma_completion data.
PR has few tests as part of gtest infrastructure.

@OphirMunk @alexvm @rosenbaumalex please look at it.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/504/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/517/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/505/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/518/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
@igor-ivanov please indicate which commits should be squashed

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/527/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/538/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
@OphirMunk it is updated with ~ring() fix

**Comment by igor-ivanov** [comment]
@OphirMunk see last commit that replace dqueue with list as https://github.com/Mellanox/libvma/pull/215/commits/1af1ea9939375d7dfefcf107a9d71fa3a002757e

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/528/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/539/ for details (Mellanox internal link).

File: src/vma/dev/ring.cpp [hunk 1]
```diff
@@ -17,11 +17,12 @@ ring::ring(int count, uint32_t mtu) :
...
 	m_vma_active(false), m_mtu(mtu)
 {
+	INIT_LIST_HEAD(&m_ec_list);
 }
 
 ring::~ring()
 {
-	while (!m_ec_queue.empty()) {
-		m_ec_queue.pop_front();
+	while (!list_empty(&m_ec_list)) {
```
Comment (OphirMunk) [inline src/vma/dev/ring.cpp]: Is a loop required here?

Comment (igor-ivanov) [inline src/vma/dev/ring.cpp]: see updated version

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/529/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/540/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/530/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/541/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/532/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/543/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/533/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/544/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/534/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/545/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/535/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/546/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/536/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/537/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
Merged #215

**Pull request closed without merging**